robot/impl/resource_manager.go

```diff
@@ -489,6 +490,11 @@ func (manager *resourceManager) completeConfig(
 ) {
 	manager.configLock.Lock()
 	defer manager.configLock.Unlock()
+	defer func() {
+		if err := manager.viz.SaveSnapshot(manager.resources); err != nil {
```

Comment:
Discussed a bit offline, but I'm curious if we should save snapshots at the end of `localRobot.Reconfigure` instead of here. Seems like we "miss" the actual node removal step, as that won't happen until `removeMarkedAndClose` gets called _after_ `completeConfig`. The disadvantage of snapshotting at the end of `localRobot.Reconfigure`, however, is that we miss potential resource graph changes that happen in [this goroutine](https://github.com/viamrobotics/rdk/blob/main/robot/impl/local_robot.go#L496). Is there a way we could get snapshots from both without getting snapshots where nothing has actually changed?

Reply:
thank you for the context and great catch! will update

Reply:
> Is there a way we could get snapshots from both without getting snapshots where nothing has actually changed?

sorry didn't address this last bit. I have some logic in `SaveSnapshot` that throws away the snapshot if it is equal to the previous one.

Reply:
ah I see your concern - if we save a snapshot in both `completeConfig` and `removeMarkedAndClose`, we get two snapshots, one for the "changing" state and one for the final state - for example, here is changing `camera1` to `camera2`:

* initial snapshot
![image](https://github.com/viamrobotics/rdk/assets/7954103/dec421c9-7313-4c76-b2f8-22bc84d33ad9)

_change to camera2_

* penultimate snapshot
![image](https://github.com/viamrobotics/rdk/assets/7954103/c96d90f9-f76b-45be-9501-1486dba8ba64)
* current snapshot
![image](https://github.com/viamrobotics/rdk/assets/7954103/5ae1c87f-d602-4e57-a774-d3ba36b82e27)

I think it's useful to see this progression, but there's definitely opportunity to merge the visualization for these kinds of transitions.

Reply:
> I think it's useful to see this progression, but there's definitely opportunity to merge the visualization for these kinds of transitions.

I also like seeing the transition and as @dgottlieb pointed out offline I think that in between state is potentially salient for users, so would be good to show it in history IMO.

> I have some logic in SaveSnapshot that throws away the snapshot if it is equal to the previous one.

Awesome thanks!